doc/sphinx/addendum/universe-polymorphism.rst

```diff
@@ -1,393 +1,444 @@
+   collapses floating universes between themselves.
+
+
+Explicit Universes
+-------------------
+
+The syntax has been extended to allow users to explicitly bind names
+to universes and explicitly instantiate polymorphic definitions.
+
+.. cmd:: Universe @ident.
+
+   In the monorphic case, this command declares a new global universe
+   named :g:`ident`, which can be referred to using its qualified name
+   as well. Global universe names live in a separate namespace. The
+   command supports the polymorphic flag only in sections, meaning the
+   universe quantification will be discharged on each section definition
+   independently. One cannot mix polymorphic and monomorphic
+   declarations in the same section.
+
+
+.. cmd:: Constraint @ident @ord @ident.
+
+   This command declares a new constraint between named universes. The
+   order relation :n:`@ord` can be one of :math:`<`, :math:`≤` or :math:`=`. If consistent, the constraint
+   is then enforced in the global environment. Like ``Universe``, it can be
+   used with the ``Polymorphic`` prefix in sections only to declare
+   constraints discharged at section closing time. One cannot declare a
+   global constraint on polymorphic universes.
+
+   .. exn:: Undeclared universe @ident.
+
+   .. exn:: Universe inconsistency.
+
+
+Polymorphic definitions
+~~~~~~~~~~~~~~~~~~~~~~~
+
 For polymorphic definitions, the declaration of (all) universe levels
 introduced by a definition uses the following syntax:
 
-\begin{coq_example*}
-Polymorphic Definition le@{i j} (A : Type@{i}) : Type@{j} := A.
-\end{coq_example*}
-\begin{coq_example}
-Print le.
-\end{coq_example}
-
-During refinement we find that $j$ must be larger or equal than $i$, as
-we are using $A : Type@{i} <= Type@{j}$, hence the generated
-constraint. At the end of a definition or proof, we check that the only
-remaining universes are the ones declared. In the term and in general in
-proof mode, introduced universe names can be referred to in
-terms. Note that local universe names shadow global universe names.
-During a proof, one can use \texttt{Show Universes} to display
-the current context of universes.
-
-Definitions can also be instantiated explicitly, giving their full instance:
-\begin{coq_example}
-Check (pidentity@{Set}).
-Universes k l.
-Check (le@{k l}).
-\end{coq_example}
+.. coqtop:: in
+
+   Polymorphic Definition le@{i j} (A : Type@{i}) : Type@{j} := A.
+
+.. coqtop:: all
+
+   Print le.
+
+During refinement we find that :g:`j` must be larger or equal than :g:`i`, as we
+are using :g:`A : Type@{i} <= Type@{j}`, hence the generated constraint. At the
+end of a definition or proof, we check that the only remaining
+universes are the ones declared. In the term and in general in proof
+mode, introduced universe names can be referred to in terms. Note that
+local universe names shadow global universe names. During a proof, one
+can use Show Universes to display the current context of universes.
```

Comment:
Missing verbatim. And is this command indexed? (Does it have a `.. cmd::`?)

Reply:
It was wrongly indexed as an exn